expression/builtin_miscellaneous.go

```diff
@@ -1046,3 +1049,79 @@ type uuidShortFunctionClass struct {
 func (c *uuidShortFunctionClass) getFunction(ctx sessionctx.Context, args []Expression) (builtinFunc, error) {
 	return nil, errFunctionNotExists.GenWithStackByArgs("FUNCTION", "UUID_SHORT")
 }
+
+type vitessHashFunctionClass struct {
+	baseFunctionClass
+}
+
+func (c *vitessHashFunctionClass) getFunction(ctx sessionctx.Context, args []Expression) (builtinFunc, error) {
+	if err := c.verifyArgs(args); err != nil {
+		return nil, err
+	}
+	argTp := args[0].GetType().EvalType()
```

Comment:
we need to validate the type of input parameter to:

1. report error errIncorrectArgs if the type is unsupported
2. provide different sigs for these args for better performance

Reply:
You are right. I checked the Vitess docs, and the only supported input is 64 bit int anyway. So I dropped the support for all the other data types and changed the function to be int64->int64. This makes code easier and needs no validation then.